### ISSUE TITLE
Ahoy.exclude_method, request and Geocoding

### ISSUE THREAD

**Issue by Bahanix** [ankane/ahoy#462]
Hello!

If we have this kind of configuration like suggested in the README.md:
```ruby
Ahoy.exclude_method = lambda do |controller, request|
  request.ip == "192.168.1.1"
  # or, in my case:
  #request.user_agent.empty?
end
```

Then the geocoding won't work anymore:
```ruby
visit = Ahoy::Visit.where(country: nil).last
Ahoy::GeocodeV2Job.perform_now(visit.visit_token, visit.ip)

# =>
Performing Ahoy::GeocodeV2Job (Job ID: <job_id>) from Async(ahoy) with arguments: "<visit_token>", "<ip>"
Error performing Ahoy::GeocodeV2Job (Job ID: <job_id>) from Async(ahoy) in 66.71ms: NoMethodError (undefined method `ip' for nil:NilClass):
<project_path>/config/initializers/ahoy.rb:6:in `block in <top (required)>'
<ahoy_gem_path>/lib/ahoy/base_store.rb:78:in `exclude_by_method?'
<ahoy_gem_path>/lib/ahoy/base_store.rb:35:in `exclude?'
<ahoy_gem_path>/lib/ahoy/tracker.rb:195:in `exclude?'
<ahoy_gem_path>/lib/ahoy/tracker.rb:70:in `geocode'
<ahoy_gem_path>/app/jobs/ahoy/geocode_v2_job.rb:24:in `perform'
...
```

Changing the initializer to:

```ruby
Ahoy.exclude_method = lambda do |controller, request|
  request && request.ip == "192.168.1.1"
end
```

Solves the issue, but I think there is a deeper issue. I guess it's normal that `GeocodeV2Job` haven't `request` in this context, but does `Ahoy::Tracker.geocode` even need to call `exclude?`? The visit should already been excluded and shouldn't trigger a call to `Ahoy::GeocodeV2Job` in this case. Wdyt?

**Comment by ankane** [comment]
Hey @Bahanix, nice find! That should probably be removed. Feel free to submit a PR.

**Comment by ankane** [comment]
Fixed in #466
